### PR TITLE
Add support for Apple touch bar icons

### PR DESCRIPTION
**Pull request by NotMyFault**
Adds support for Apple's touch bar icons and drops obsolete `shortcut` tag, as per the MDN docs.

### Proposed changelog entries

* Add support for Apple's touch bar icons.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6768"></a>

### CODE REVIEW

**Comment by timja** [comment]
Oh cool I didn't know this was a thing, good info in https://medium.com/swlh/are-you-using-svg-favicons-yet-a-guide-for-modern-browsers-836a6aace3df

(not tested yet)

**Comment by daniel-beck** [comment]
What's the favicon used on pages not using `l:layout`? Would that still be `.ico`?

**Comment by daniel-beck** [comment]
`/env-vars.html/` before https://github.com/jenkinsci/jenkins/pull/6096 comes to mind. Given how recent that change is, it's reasonable to expect there to be more such pages.

A simple (but perhaps not _that_ relevant) example in current core might be directly accessed help pages, like https://ci.jenkins.io/help/system-config/homeDirectory.html

**Comment by NotMyFault** [comment]
> /env-vars.html/ before https://github.com/jenkinsci/jenkins/pull/6096 comes to mind. Given how recent that change is, it's reasonable to expect there to be more such pages.

This page picks up the favicon replacement fine.

> A simple (but perhaps not that relevant) example in current core might be directly accessed help pages, like https://ci.jenkins.io/help/system-config/homeDirectory.html

No favicon here. I didn't know about this page before, but it looks quite ancient. If not available, a more appealing documentation may be available on jenkins.io 👀

**Comment by timja** [comment]
> No favicon here. I didn't know about this page before, but it looks quite ancient.

Those pages are just the help pages themselves that are displayed inline, you can directly access them via URL.
They would never normally be accessed directly, not a real test really.

**Comment by daniel-beck** [comment]
I'm not going to look for a plugin that renders some pages without `l:layout`. I would be very surprised if there isn't.

This PR, for no reason at all, makes the UI worse in some (rare) situations. Why is that not enough?

**Comment by daniel-beck** [comment]
Out of curiosity, where do RSS feeds get their favicon from?

File: core/src/main/java/hudson/security/ChainedServletFilter.java
Comment (daniel-beck) [inline core/src/main/java/hudson/security/ChainedServletFilter.java:77]:
```suggestion
    private static final Pattern UNINTERESTING_URIS = Pattern.compile("/(images|jsbundles|css|scripts|adjuncts)/|/favicon[.](ico|svg)|/ajax");
```

File: core/src/main/java/jenkins/security/ResourceDomainFilter.java
Comment (daniel-beck) [inline core/src/main/java/jenkins/security/ResourceDomainFilter.java:57]:
```suggestion
    private static final Set<String> ALLOWED_PATHS = new HashSet<>(Arrays.asList("/" + ResourceDomainRootAction.URL, "/favicon.ico", "/favicon.svg", "/robots.txt"));
```

**Review by daniel-beck — COMMENTED** [review]
Looks like these would also need restoring?

IIRC, the favicon.ico is allowed on resource root URL so that browsers that request it don't spam all possible logs with errors (and would apply to plain text files etc. as well). That got annoying quickly.

**Comment by NotMyFault** [comment]
> Looks like these would also need restoring?

I committed your suggestions, are there outstanding concerns left?

**Review by timja — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
